src/routes/mod_creation.rs

```diff
@@ -0,0 +1,401 @@
+                                    );
+
+                                    upload_file(
+                                        upload_url.get_ref().clone(),
+                                        "text/plain".to_string(),
+                                        body_url.clone(),
+                                        version_data.version_body.into_bytes(),
+                                    )
+                                    .await?;
+
+                                    let upload_data = upload_file(
+                                        upload_url.get_ref().clone(),
+                                        "application/java-archive".to_string(),
+                                        format!(
+                                            "{}/{}/{}",
+                                            create_data.mod_namespace.replace(".", "/"),
+                                            version_data.version_slug,
+                                            file_name
+                                        ),
+                                        (&data).to_owned().to_vec(),
+                                    )
+                                    .await?;
+
+                                    let version = Version {
+                                        version_id: version_id.0 as i32,
+                                        mod_id: mod_id.0 as i32,
+                                        name: version_data.version_title,
+                                        number: version_data.version_slug.clone(),
+                                        changelog_url: Some(format!("{}/{}", cdn_url, body_url)),
+                                        date_published: Utc::now().to_rfc2822(),
+                                        downloads: 0,
+                                        version_type: version_data.version_type.to_string(),
+                                        files: vec![VersionFile {
+                                            game_versions: version_data
+                                                .game_versions
+                                                .into_iter()
+                                                .map(|x| x.0)
+                                                .collect::<Vec<_>>(),
+                                            hashes: vec![FileHash {
+                                                algorithm: "sha1".to_string(),
+                                                hash: upload_data.content_sha1,
+                                            }],
+                                            url: format!("{}/{}", cdn_url, upload_data.file_name),
+                                        }],
+                                        dependencies: version_data
+                                            .dependencies
+                                            .into_iter()
+                                            .map(|x| x.0 as i32)
```

Comment:
This is currently reinterpreting the input `ModId` (which should have been a `VersionId`) as a `VersionId`.  Be careful about that - in this place the other location is incorrect, but it could happen elsewhere.